matchmaking/src/authenticate.py

```diff
@@ -0,0 +1,42 @@
+from typing import Optional
+
+import jwt
+import requests
+
+import src.settings as settings
+
+
+def authenticate_user(token: str) -> Optional[dict]:
+    payload = decode_jwt(token)
+    if payload is None:
+        return None
+    user_id = payload.get('user_id')
+    if user_id is None:
+        return None
+    # TODO: Uncomment the following lines when the `user/` endpoint is implemented
+    # user = get_user(user_id, token)
+    # return user
+    return payload
+
+
+def decode_jwt(token: str) -> Optional[dict]:
+    try:
+        payload = jwt.decode(
+            token,
+            settings.VALID_TOKEN,
+            algorithms=[settings.DECODE_ALGORITHM],
+        )
+        return payload
+    except Exception:
+        return None
+
+
+def get_user(user_id: int, token: str) -> Optional[dict]:
+    headers = {'Authorization': token}
+    try:
+        response = requests.get(f'http://user-management-nginx/user/{user_id}/', headers=headers)
```

Comment:
Maybe use URL variable for every request on user-management ?

Reply:
7f92e5c

Moved user-management in settings, we should use common folder in the future, see issue https://github.com/tdameros/42-transcendence/issues/34